src/main/java/com/gl/donate_receive/service/converter/ReportConverter.java

```diff
@@ -2,15 +2,45 @@
 
 import com.gl.donate_receive.dto.ReportDto;
 import com.gl.donate_receive.model.Report;
+import com.gl.donate_receive.model.ReportId;
 import org.springframework.stereotype.Component;
+import org.springframework.web.multipart.MultipartFile;
+
+import java.io.IOException;
+import java.util.UUID;
 
 @Component
 public class ReportConverter {
 
-    public Report dtoToModel(ReportDto reportDto) {
-        return Report.builder()
-                .comment(reportDto.getComment())
-                .mediaFile(reportDto.getMediaFile())
-                .build();
-    }
+	public Report dtoToModel(ReportDto reportDto) {
+		var reportId = getReportId(reportDto.getReceiverId(), reportDto.getReceiverId());
+		return Report.builder()
+			.comment(reportDto.getComment())
+			.reportId(reportId)
+			.mediaFile(convertMediaFileToBytesArray(reportDto.getFile()))
+			.build();
+	}
+
+	private ReportId getReportId(String itemId, String receiverId) {
+		var reportId = new ReportId();
+		reportId.setItemId(UUID.fromString(itemId));
+		reportId.setReceiverId(UUID.fromString(receiverId));
+		return reportId;
+	}
+
+	private byte[] convertMediaFileToBytesArray(MultipartFile file) {
+		if (file == null) {
+			return new byte[0];
+		}
+
+		final byte[] mediaFile;
+		try {
+			mediaFile = file.getBytes();
+		} catch (IOException e) {
+			throw new IllegalArgumentException("Cannot parse received file");

```

Comment:
What are the conditions of this exception?

Reply:
From https://docs.spring.io/:
Throws:
[IOException](https://docs.oracle.com/javase/8/docs/api/java/io/IOException.html?is-external=true) - in case of access errors (if the temporary store fails)